src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -220,72 +182,135 @@ public void canCalculateEffectiveLocationOnHoldingUpdateWhenSomeItemsHasLocation
 
     // Assert that itemWithPermLocationFetched was not updated
     assertEquals(itemWithTempLocationFetched.getEffectiveLocationId(), annexLibraryLocationId.toString());
-
   }
 
-  @Test
-  public void canCalculateEffectiveLocationOnItemPermLocationUpdate() throws Exception {
-    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
-
-    Item item = buildItem(holdingsRecordId, null, null);
-    createItem(item);
+  private void setPermTempLocation(JsonObject json, int perm, int temp) {
+    if (perm == 0) {
+      json.remove("permanentLocationId");
+    } else {
+      json.put("permanentLocationId", location[perm].toString());
+    }
+    if (temp == 0) {
+      json.remove("temporaryLocationId");
+    } else {
+      json.put("temporaryLocationId", location[temp].toString());
+    }
+  }
 
-    Item itemFetched = getItem(item.getId());
-    assertEquals(itemFetched.getEffectiveLocationId(), annexLibraryLocationId.toString());
+  private String effectiveLocation(int holdingPerm, int holdingTemp, int itemPerm, int itemTemp) {
+    int loc = 0;
+    if (itemTemp != 0) {
```

Comment:
Is this determining the effective location within the tests?

I prefer that tests do not use conditional parts as it makes them harder to follow and understand, and hence less effective at describing the intended behaviour.

Reply:
outdated

Reply:
Yes, we calculate the expected effective location in the test code.
It does not make sense to put a hardcoded list of all location string combinations and the expected effective location into the test class.
We want to test the triggers. The triggers have complicated code to avoid as many additional SELECT queries for performance reasons. We want to make sure that this complicated code comes to the correct result.